File: java-spring/crud/src/main/java/com/example/crud/entity/Feed.java
package com.example.crud.entity;

import com.example.crud.event.FeedEntityListener;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(value = FeedEntityListener.class)
@Entity
public class Feed {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "BIGINT", nullable = false)
    private Long id;

    @NotNull
    @Column(columnDefinition = "VARCHAR(20)")
    private String title;

    @NotNull
    @Column(columnDefinition = "VARCHAR(1000)")
    private String content;

    @Builder
    public Feed(String title, String content) {
        this.title = title;
        this.content = content;
    }

    public void update(String title, String content) {
        this.title = title;
        this.content = content;
    }
}

File: kotlin-spring/crud/src/main/kotlin/com/example/crud/entity/Feed.kt
package com.example.crud.entity

import javax.persistence.*

@Entity
class Feed (
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        private val id : Long = 0,

        @Column(columnDefinition = "VARCHAR(20)", nullable = false)
        private val title : String,

        @Column(columnDefinition = "VARCHAR(1000)", nullable = false)
        private val content : String,
) {

}